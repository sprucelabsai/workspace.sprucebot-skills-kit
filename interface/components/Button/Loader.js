import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class Loader extends Component {
	render() {
		const { blue, fullWidth } = this.props
		const dotClassName = blue ? 'loader_dot_blue' : 'loader_dot'
		const fullWidthStyle = fullWidth
			? { display: 'block', margin: '20px', textAlign: 'center' }
			: {}

		return (
			<span className="loader_wrapper" style={fullWidthStyle}>
				<span className={dotClassName} />
				<span className={dotClassName} />
				<span className={dotClassName} />
			</span>
		)
	}
}

Loader.propTypes = {
	blue: PropTypes.bool,
	fullWidth: PropTypes.bool
}

Loader.defaultProps = {
	blue: true,
	fullWidth: true
}
