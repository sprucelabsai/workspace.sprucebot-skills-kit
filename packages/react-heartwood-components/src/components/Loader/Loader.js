// @flow
import React from 'react'
import cx from 'classnames'

type Props = {
	/* set to true if loader is on a dark background */
	isLight: boolean,

	/* set to true if loader should be centered within parent */
	isCentered: boolean
}

const Loader = (props: Props) => {
	const { isLight, isCentered, ...rest } = props
	return (
		<span
			className={cx('loader', {
				'loader--light': isLight,
				'loader--centered': isCentered
			})}
			{...rest}
		>
			<span className="loader__dot" />
			<span className="loader__dot" />
			<span className="loader__dot" />
		</span>
	)
}

Loader.defaultProps = {
	isLight: false
}

export default Loader
