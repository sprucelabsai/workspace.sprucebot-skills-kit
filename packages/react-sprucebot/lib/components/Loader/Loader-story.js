"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _react2 = require("@storybook/react");

var _Loader = _interopRequireDefault(require("./Loader"));

(0, _react2.storiesOf)('Loader', module).add('Loader', function () {
  return _react.default.createElement(_Loader.default, null);
});