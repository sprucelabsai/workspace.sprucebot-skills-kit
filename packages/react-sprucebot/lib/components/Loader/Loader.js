"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var Loader = function Loader() {
  return _react.default.createElement("span", {
    className: "loader"
  }, _react.default.createElement("span", {
    className: "loader__dot"
  }), _react.default.createElement("span", {
    className: "loader__dot"
  }), _react.default.createElement("span", {
    className: "loader__dot"
  }));
};

var _default = Loader;
exports.default = _default;