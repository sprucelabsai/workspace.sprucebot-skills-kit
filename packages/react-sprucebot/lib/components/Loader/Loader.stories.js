'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _Loader = require('./Loader');

var _Loader2 = _interopRequireDefault(_Loader);

var _Loader3 = require('./Loader.md');

var _Loader4 = _interopRequireDefault(_Loader3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('Loader', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_Loader4.default, function () {
  return _react2.default.createElement(_Loader2.default, null);
}));