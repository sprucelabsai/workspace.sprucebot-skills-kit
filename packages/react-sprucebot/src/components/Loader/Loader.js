import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class Loader extends Component {
	render() {
		const { dark, fullWidth } = this.props
		const dotClassName = dark ? 'loader_dot_dark' : 'loader_dot'
		const fullWidthStyle = fullWidth
			? { display: 'block', margin: '20px', textAlign: 'center' }
			: {}

		return (
			<span className="loader_wrapper" style={fullWidthStyle}>
				<span className={dotClassName} />
				<span className={dotClassName} />
				<span className={dotClassName} />
			</span>
		)
	}
}

Loader.propTypes = {
	dark: PropTypes.bool,
	fullWidth: PropTypes.bool
}

Loader.defaultProps = {
	dark: true,
	fullWidth: true
}
