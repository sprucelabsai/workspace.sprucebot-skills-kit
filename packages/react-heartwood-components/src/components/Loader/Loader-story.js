import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs, boolean } from '@storybook/addon-knobs/react'
import Loader from './Loader'

const stories = storiesOf('Loader', module)

stories.addDecorator(withKnobs)

stories.add('Loader', () => {
	const isLight = boolean('isLight', false)
	return (
		<div
			style={{
				backgroundColor: isLight ? 'black' : 'transparent',
				padding: '2rem',
				width: '100%',
				height: '100%',
				position: 'absolute'
			}}
		>
			<Loader isLight={isLight} isCentered={boolean('isCentered', false)} />
		</div>
	)
})
