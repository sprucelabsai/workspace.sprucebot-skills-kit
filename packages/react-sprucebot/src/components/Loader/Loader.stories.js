import React from 'react'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withReadme } from 'storybook-readme'
import Loader from './Loader'
import readme from './Loader.md'

const stories = storiesOf('Loader', module)
stories.addDecorator(withKnobs)

stories.add('Interactive', withReadme(readme, () => <Loader />))
