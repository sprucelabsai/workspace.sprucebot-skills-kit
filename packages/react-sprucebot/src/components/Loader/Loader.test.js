import React from 'react'
import Loader from './Loader'
import { shallow } from 'enzyme'

describe('Loader tests', () => {
	it('Should match the snapshot', () => {
		const renderedComponent = shallow(<Loader />)

		expect(renderedComponent).toMatchSnapshot()
	})
})
