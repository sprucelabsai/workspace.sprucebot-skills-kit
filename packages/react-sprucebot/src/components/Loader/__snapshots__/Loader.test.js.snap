// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  display: block;
  margin: 20px;
  text-align: center;
}

<span
  className="loader_wrapper c0"
  style={Object {}}
>
  <span
    className="loader_dot_dark"
  />
  <span
    className="loader_dot_dark"
  />
  <span
    className="loader_dot_dark"
  />
</span>
`;
